collated/main/fustilio.md

```diff
@@ -0,0 +1,581 @@
+     * to keep track of which tab the model is "on".
+     */
+    @Subscribe
+    private void handleJumpToTabEvent(JumpToTabRequestEvent event) {
+        model.setTabIndex(Index.fromZeroBased(event.targetIndex));
+    }
+}
+```
+###### \java\seedu\address\model\ModelManager.java
+``` java
+    /** Deletes the tag from every parcel in the address book */
+    public void deleteTag(Tag target) throws TagNotFoundException, TagInternalErrorException {
+
+        int tagsFound = 0;
+        Iterator it = addressBook.getParcelList().iterator();
+        while (it.hasNext()) {
+            Parcel oldParcel = (Parcel) it.next();
+            Parcel newParcel = new Parcel(oldParcel);
+            Set<Tag> newTags = new HashSet<>(newParcel.getTags());
+            if (newTags.contains(target)) {
+                newTags.remove(target);
+                tagsFound++;
+            }
+
+            newParcel.setTags(newTags);
+
+            try {
+                addressBook.updateParcel(oldParcel, newParcel);
+            } catch (DuplicateParcelException | ParcelNotFoundException dpe) {
+                throw new TagInternalErrorException();
+            }
+        }
+
+        if (tagsFound == 0) {
+            throw new TagNotFoundException();
+        }
+
+        indicateAddressBookChanged();
+    }
+```
+###### \java\seedu\address\model\ModelManager.java
+``` java
+    @Override
+    public void maintainSorted() {
+        addressBook.sort();
+        indicateAddressBookChanged();
+    }
+
+    @Override
+    public void forceSelect(Index target) {
+        EventsCenter.getInstance().post(new JumpToListRequestEvent(target));
+    }
+
+    @Override
+    public void forceSelectParcel(ReadOnlyParcel target) {
+        forceSelect(Index.fromZeroBased(findIndex(target)));
+    }
+
+    @Override
+    public void setTabIndex(Index index) {
+        this.tabIndex = index;
+    }
+
+    @Override
+    public Index getTabIndex() {
+        return this.tabIndex;
+    }
+
+    @Override
+    public void addParcelCommand(ReadOnlyParcel toAdd) throws DuplicateParcelException {
+        this.addParcel(toAdd);
+        this.maintainSorted();
+        this.handleTabChange(toAdd);
+        this.forceSelectParcel(toAdd);
+        indicateAddressBookChanged();
+    }
+
+    @Override
+    public void editParcelCommand(ReadOnlyParcel parcelToEdit, ReadOnlyParcel editedParcel)
+            throws DuplicateParcelException, ParcelNotFoundException {
+        this.updateParcel(parcelToEdit, editedParcel);
+        this.maintainSorted();
+        this.handleTabChange(editedParcel);
+        this.forceSelectParcel(editedParcel);
+        indicateAddressBookChanged();
+    }
+
+    @Override
+    public boolean getActiveIsAllBool() {
+        return tabIndex.equals(TAB_ALL_PARCELS);
+    }
+
+    /**
+     * Method to internally change the active list to the correct tab according to the changed parcel.
+     * @param targetParcel
```

Comment:
For @param If the description is self explanatory then it can be omitted.
Do not leave javadoc tags without a description.